sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefService.java

```diff
@@ -112,32 +108,31 @@ public List<SclReportItem> updateAllExtRefIedNames(SCL scd) {
     }
 
     @Override
-    public List<SclReportItem> manageBindingForLDEPF(SCL scd, ILDEPFSettings settings) {
-        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+    public List<SclReportItem> manageBindingForLDEPF(SCL scd, EPF epf) {
```

Comment:
We should try to not use adapters when it's possible, it will save us time when we will refactor the code : 
```java
@Override
    public List<SclReportItem> manageBindingForLDEPF(SCL scd, EPF epf) {
        List<SclReportItem> sclReportItems = new ArrayList<>();
        if (!epf.isSetChannels()) return List.of();

        scd.getIED()
                .stream()
                .filter(tied -> !IED_TEST_NAME.equals(tied.getName()))
                .filter(TIED::isSetAccessPoint)
                .flatMap(tied -> tied.getAccessPoint().stream())
                .map(TAccessPoint::getServer)
                .filter(Objects::nonNull)
                .filter(TServer::isSetLDevice)
                .flatMap(tServer -> tServer.getLDevice().stream())
                .filter(tlDevice -> LDEVICE_LDEPF.equals(tlDevice.getInst()))
                .map(tlDevice -> new LDeviceAdapter(null, tlDevice))
                .forEach(lDeviceAdapter -> lDeviceAdapter.getExtRefBayReferenceForActifLDEPF(sclReportItems)
                        .forEach(extRefBayRef -> epf.getChannels()
                                .getChannel()
                                .stream()
                                .filter(tChannel -> isExtRefMatchLDEPFChannel(extRefBayRef.extRef(), tChannel))
                                .findFirst()
                                .filter(tChannel -> TCBscopeType.BAY_EXTERNAL.equals(tChannel.getBayScope()))
                                .ifPresent(channel -> {
                                    List<TIED> iedSources = getIedSources(new SclRootAdapter(scd), extRefBayRef.compasBay(), channel);
                                    if (iedSources.size() == 1) {
                                        updateLDEPFExtRefBinding(extRefBayRef.extRef(), iedSources.get(0), channel);
                                        sclReportItems.addAll(updateLDEPFDos(lDeviceAdapter, extRefBayRef.extRef(), channel));
                                    } else {
                                        if (iedSources.size() > 1) {
                                            sclReportItems.add(SclReportItem.warning(null, "There is more than one IED source to bind the signal " +
                                                    "/IED@name=" + extRefBayRef.iedName() + "/LDevice@inst=LDEPF/LN0" +
                                                    "/ExtRef@desc=" + extRefBayRef.extRef().getDesc()));
                                        }
                                    }
                                })));
        return sclReportItems;
    }
```

Reply:
Actually, I can't precede this request, for many reason